volto/i18n.rst

```diff
@@ -0,0 +1,242 @@
+    Synchronizing messages to po files...
+    Generating the json files...
+    done!
+
+As the description suggests it will first extract all messages from the source
+files into a json file. Then it will synchronize the extracted messages with the
+.pot master file and with all the .po files found in the project. After that it
+will generate the final .json files which are used in the application. If you
+want to translate the strings just edit the .po files and run the script again.
+This script will combine the messages located in Volto itself and the current
+project, and combine them into the json files.
+
+Excercise
+=========
+
+Translate the "Tags:" label which was added in the previous chapter to dutch and
+german.
+
+..  admonition:: Solution
+    :class: toggle
+
+    ::
+
+        /**
+         * Tags component.
+         * @module components/theme/Tags/Tags
+         */
+
+        import React from 'react';
+        import { Link } from 'react-router-dom';
+        import PropTypes from 'prop-types';
+        import { Container } from 'semantic-ui-react';
+        import { FormattedMessage } from 'react-intl';
+
+        /**
+         * Tags component class.
+         * @function Tags
+         * @param {array} tags Array of tags.
+         * @returns {string} Markup of the component.
+         */
+        const Tags = ({ tags }) =>
+          tags && tags.length > 0 ? (
+            <Container>
+              <FormattedMessage id="Tags" defaultMessage="Tags" />:
+              {tags.map(tag => (
+                <Link className="ui label" to={`/search?Subject=${tag}`} key={tag}>
+                  {tag}
+                </Link>
+              ))}
+            </Container>
+          ) : (
+            <span />
+          );
+
+        /**
+         * Property types.
+         * @property {Object} propTypes Property types.
+         * @static
+         */
+        Tags.propTypes = {
+          tags: PropTypes.arrayOf(PropTypes.string),
+        };
+
+        /**
+         * Default properties.
+         * @property {Object} defaultProps Default properties.
+         * @static
+         */
+        Tags.defaultProps = {
+          tags: null,
+        };
+
+        export default Tags;
+
+Translate Attributes
+====================
+
+When you want to translate attributes you can't use the :file:`FormattedMessage`
+component since it will generate markup. In order to translate a normal string
+or attribute you can use the :file:`formatMessage` method of :file:`react-intl`.
+First you need to import all the required methods:
+
+::
+
+    import { defineMessages, injectIntl, intlShape } from 'react-intl';
+
+Then you can use the :file:`defineMessages` method to define your messages:
+
+::
+
+    const messages = defineMessages({
+      site: {
+        id: 'Site',
+        defaultMessage: 'Site',
+      },
+    });
+
+In order to to use the :file:`formatMessage` method you have to inject the intl
```

Comment:
"intl"? What is this? Should this be spelled out as "international" or is it inline code that requires double backticks?

```
``intl``
``